sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuApproximatePercentile.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import ai.rapids.cudf.{ColumnVector, RollingAggregationOnColumn}
+import com.nvidia.spark.rapids.GpuCast.{recursiveDoColumnar}
+
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, Expression}
+import org.apache.spark.sql.catalyst.expressions.aggregate.ApproximatePercentile
+import org.apache.spark.sql.catalyst.util.ArrayData
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.rapids.{CudfTDigest, GpuAggregateFunction}
+import org.apache.spark.sql.types.{ArrayType, DataType, DataTypes}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+case class GpuApproximatePercentile (
+    child: Expression,
+    percentageExpression: GpuLiteral,
+    accuracyExpression: GpuLiteral)
+  extends GpuAggregateFunction with GpuAggregateWindowFunction {
+
+  override val inputProjection: Seq[Expression] = Seq(child)
+
+  /** Attributes of fields in aggBufferSchema. */
+  override def aggBufferAttributes: Seq[AttributeReference] = outputBuf :: Nil
+
+  // Mark as lazy to avoid being initialized when creating a GpuApproximatePercentile.
+  override lazy val initialValues: Seq[GpuExpression] = throw new UnsupportedOperationException
+
+  override lazy val updateExpressions: Seq[Expression] =
+    new CudfTDigest(inputBuf,
+      percentageExpression,
+      accuracyExpression) :: Nil
+
+  override lazy val mergeExpressions: Seq[GpuExpression] =
```

Comment:
Sorry I'm a bit confused. Here are the original methods from `GpuAggregateFunctions`:
```
// these are values that spark calls initial because it uses
  // them to initialize the aggregation buffer, and returns them in case
  // of an empty aggregate when there are no expressions,
  // here we copy them but with the gpu equivalent
  val initialValues: Seq[GpuExpression]

  // update: first half of the aggregation (count = count)
  val updateExpressions: Seq[Expression]

  // merge: second half of the aggregation (count = sum). Also use to merge multiple batches.
  val mergeExpressions: Seq[GpuExpression]

  // mostly likely a pass through (count => sum we merged above).
  // average has a more interesting expression to compute the division of sum/count
  val evaluateExpression: Expression
```

So we are using mixed `GpuExpression` and `Expression`. Why?

Reply:
This is updated now to use `Seq[Expression]` in both cases